### PR TITLE
Gem

### PR DESCRIPTION
**Pull request by jonathanmorley**
Turn the project into a functioning gem with binstubs

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -6,7 +6,13 @@ init:
...
-  - ps: ./bootstrap.ps1
-
-test_script:
-  - ps: chef exec inspec exec test/integration/**/*.rb --format documentation
+  - ps: . { (New-Object System.Net.WebClient).DownloadString('https://omnitruck.chef.io/install.ps1') } | iex; install -project 'chefdk' -version '0.17.17'
+  - ps: $env:path += ';C:\opscode\chefdk\bin;'
+  - ps: chef exec bundle install
+  - ps: chef exec bundle exec rake
+  - ps: chef gem build ./workup.gemspec
+  - ps: chef gem install ./workup-0.1.0.gem --bindir 'C:\opscode\chefdk\bin'
```
Comment (brentm5) [inline .appveyor.yml]: use rake to do this

File: .appveyor.yml [hunk 2]
```diff
@@ -6,7 +6,13 @@ init:
   - ps: $PSVersionTable
 
 build_script:
-  - ps: ./bootstrap.ps1
-
-test_script:
-  - ps: chef exec inspec exec test/integration/**/*.rb --format documentation
+  - ps: . { (New-Object System.Net.WebClient).DownloadString('https://omnitruck.chef.io/install.ps1') } | iex; install -project 'chefdk' -version '0.17.17'
```
Comment (brentm5) [inline .appveyor.yml]: Why not use the default way to install chefdk instead of this way?

File: .appveyor.yml [hunk 3]
```diff
@@ -6,7 +6,13 @@ init:
...
+  - ps: $env:path += ';C:\opscode\chefdk\bin;'
+  - ps: chef exec bundle install
+  - ps: chef exec bundle exec rake
+  - ps: chef gem build ./workup.gemspec
+  - ps: chef gem install ./workup-0.1.0.gem --bindir 'C:\opscode\chefdk\bin'
+  - ps: mkdir ~/.workup
+  - ps: cp ./files/Policyfile.rb ~/.workup/Policyfile.rb
```
Comment (brentm5) [inline .appveyor.yml]: I would just make a folder in the test folder to mock this stuff and move the entire folder to ~/.workup

File: .travis.yml [hunk 4]
```diff
@@ -2,6 +2,25 @@ os:
...
+before_install:
+  - gem update bundler
+
 script:
-  - sudo ./bootstrap.sh
-  - chef exec inspec exec test/integration/**/*.rb --format documentation
+  - curl -L https://omnitruck.chef.io/install.sh | sudo bash -s -- -P 'chefdk' -v "0.17.17"
+  - chef exec bundle install
+  - chef exec bundle exec rake
+  - chef gem build ./workup.gemspec
```
Comment (brentm5) [inline .travis.yml]: chef exec bundle exec rake install

File: .travis.yml [hunk 5]
```diff
@@ -2,6 +2,25 @@ os:
...
+  - sudo chef gem install ./workup-0.1.0.gem --bindir '/usr/local/bin'
+  - mkdir ~/.workup
```
Comment (brentm5) [inline .travis.yml]: Ditto

File: .travis.yml [hunk 6]
```diff
@@ -2,6 +2,25 @@ os:
...
+  only:
+    - master
+
+before_install:
+  - gem update bundler
+
 script:
-  - sudo ./bootstrap.sh
-  - chef exec inspec exec test/integration/**/*.rb --format documentation
+  - curl -L https://omnitruck.chef.io/install.sh | sudo bash -s -- -P 'chefdk' -v "0.17.17"
```
Comment (brentm5) [inline .travis.yml]: I would move this to the before_install script

File: files/Policyfile.rb [hunk 7]
```diff
@@ -2,6 +2,9 @@
 
 default_source :community
 
+cookbook 'no_op', github: 'livinginthepast/windows-noop-cookbook'
```
Comment (brentm5) [inline files/Policyfile.rb]: WHy not 

https://supermarket.chef.io/cookbooks/nop

File: .appveyor.yml (hunk 1 above)
Comment (jonathanmorley) [inline .appveyor.yml]: This wouldn't install it with the correct bindir though

File: .appveyor.yml (hunk 2 above)
Comment (jonathanmorley) [inline .appveyor.yml]: Its basically the same, this is emulating the bootstrap script but locally, so closer to what will happen

File: .appveyor.yml (hunk 3 above)
Comment (jonathanmorley) [inline .appveyor.yml]: Good idea

File: .travis.yml (hunk 6 above)
Comment (jonathanmorley) [inline .travis.yml]: Sure, just trying to figure out why its not working right now

File: files/Policyfile.rb (hunk 7 above)
Comment (jonathanmorley) [inline files/Policyfile.rb]: Because i didn't find it in my search. Will replace with `nop`

**Comment by brentm5** [comment]
Rebase before you merge

**Pull request closed without merging**